recipes/vigra/build.sh

```diff
@@ -0,0 +1,79 @@
+    export LIBRARY_SEARCH_VAR=DYLD_FALLBACK_LIBRARY_PATH
+    export MACOSX_VERSION_MIN="10.7"
+    export CMAKE_OSX_DEPLOYMENT_TARGET="${MACOSX_VERSION_MIN}"
+    export CXXFLAGS="-mmacosx-version-min=${MACOSX_VERSION_MIN}"
+    export CXXFLAGS="${CXXFLAGS} -stdlib=libc++ -std=c++11"
+    export LINKFLAGS="-mmacosx-version-min=${MACOSX_VERSION_MIN}"
+    export LINKFLAGS="${LINKFLAGS} -stdlib=libc++ -std=c++11 -L${LIBRARY_PATH}"
+    EXTRA_CMAKE_ARGS="-DCMAKE_OSX_DEPLOYMENT_TARGET=${CMAKE_OSX_DEPLOYMENT_TARGET}"
+else
+    DYLIB_EXT=so
+    export LIBRARY_SEARCH_VAR=LD_LIBRARY_PATH
+    export CXXFLAGS="-pthread -std=c++11 ${CXXFLAGS}"
+fi
+export EXTRA_CMAKE_ARGS
+
+export VIGRA_CXX_FLAGS="${CXXFLAGS}"
+
+# In release mode, we use -O2 because gcc is known to miscompile certain vigra functionality at the O3 level.
+# (This is probably due to inappropriate use of undefined behavior in vigra itself.)
+export VIGRA_CXX_FLAGS_RELEASE="-O2 -DNDEBUG ${VIGRA_CXX_FLAGS}"
+export VIGRA_LDFLAGS="${CXX_LDFLAGS} -Wl,-rpath,${PREFIX}/lib -L${PREFIX}/lib"
+
+# Configure
+mkdir build
+cd build
+cmake ..\
+        -DCMAKE_INSTALL_PREFIX=${PREFIX} \
+        -DCMAKE_PREFIX_PATH=${PREFIX} \
+\
+        -DCMAKE_CXX_LINK_FLAGS="${VIGRA_LDFLAGS}" \
+        -DCMAKE_EXE_LINKER_FLAGS="${VIGRA_LDFLAGS}" \
+        -DCMAKE_CXX_FLAGS="${VIGRA_CXX_FLAGS}" \
+        -DCMAKE_CXX_FLAGS_RELEASE="${VIGRA_CXX_FLAGS_RELEASE}" \
+        -DCMAKE_CXX_FLAGS_DEBUG="${VIGRA_CXX_FLAGS}" \
+\
+        -DWITH_VIGRANUMPY=TRUE \
+        -DWITH_BOOST_THREAD=1 \
+        -DDEPENDENCY_SEARCH_PREFIX=${PREFIX} \
+\
+        -DFFTW3F_INCLUDE_DIR=${PREFIX}/include \
+        -DFFTW3F_LIBRARY=${PREFIX}/lib/libfftw3f.${DYLIB_EXT} \
+        -DFFTW3_INCLUDE_DIR=${PREFIX}/include \
+        -DFFTW3_LIBRARY=${PREFIX}/lib/libfftw3.${DYLIB_EXT} \
+\
+        -DHDF5_CORE_LIBRARY=${PREFIX}/lib/libhdf5.${DYLIB_EXT} \
+        -DHDF5_HL_LIBRARY=${PREFIX}/lib/libhdf5_hl.${DYLIB_EXT} \
+        -DHDF5_INCLUDE_DIR=${PREFIX}/include \
+\
+        -DBoost_INCLUDE_DIR=${PREFIX}/include \
+        -DBoost_LIBRARY_DIRS=${PREFIX}/lib \
+\
+        -DPYTHON_EXECUTABLE=${PYTHON} \
+        -DPYTHON_INCLUDE_PATH=${PREFIX}/include \
+\
+        -DZLIB_INCLUDE_DIR=${PREFIX}/include \
+        -DZLIB_LIBRARY=${PREFIX}/lib/libz.${DYLIB_EXT} \
+\
+        -DPNG_LIBRARY=${PREFIX}/lib/libpng.${DYLIB_EXT} \
+        -DPNG_PNG_INCLUDE_DIR=${PREFIX}/include \
+\
+        -DTIFF_LIBRARY=${PREFIX}/lib/libtiff.${DYLIB_EXT} \
+        -DTIFF_INCLUDE_DIR=${PREFIX}/include \
+\
+        -DJPEG_INCLUDE_DIR=${PREFIX}/include \
+        -DJPEG_LIBRARY=${PREFIX}/lib/libjpeg.${DYLIB_EXT} \
+        ${EXTRA_CMAKE_ARGS}
+
+make
+# Can't run tests due to a bug in the clang compiler provided with XCode.
+# For more details see here ( https://llvm.org/bugs/show_bug.cgi?id=21083 ).
+# Also, these tests are very intensive, which makes them challenging to run in CI.
+#eval ${LIBRARY_SEARCH_VAR}=$PREFIX/lib make check
```

Comment:
Unfortunately, we have to disable `make check`. There are two reasons.

1. Building and running the tests takes a long time so is impractical on CI when trying a matrix of Python and NumPy versions in the same build. Once these get split out it may be possible again.
2. The version of `clang` that we are using with the copy of XCode on the Mac VM has a [bug]( https://llvm.org/bugs/show_bug.cgi?id=21083 ). It has been fixed in later versions of `clang`, but those versions are available with copies of XCode that support 10.7.